ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/application/plugins/api/ApplicationPluginBuilder.kt

```diff
@@ -16,43 +14,13 @@ import io.ktor.util.debug.*
 import io.ktor.util.pipeline.*
 import kotlin.random.*
 
-/**
- * A builder that is available inside a plugin creation block. It allows you to define handlers for different stages
- * (a.k.a. phases) of the HTTP pipeline.
- **/
-public interface PluginBuilderBase {
-    /**
-     * Specifies how to modify HTTP call handling for the current [PluginBuilder].
-     * @see OnCall
-     **/
-    public val onCall: OnCall
-
-    /**
-     * Specifies how to modify receiving data from an HTTP call for the current [PluginBuilder].
-     * @see OnCallReceive
-     **/
-    public val onCallReceive: OnCallReceive
-
-    /**
-     * Specifies how to modify sending data within an HTTP call for the current [PluginBuilder].
-     * @see OnCallRespond
-     **/
-    public val onCallRespond: OnCallRespond
-
-    /**
-     * Specifies a shutdown hook. This method is useful for closing resources allocated by the plugin.
-     *
-     * @param hook An action that needs to be executed when the application shuts down.
-     **/
-    public fun applicationShutdownHook(hook: (Application) -> Unit)
-}
-
 /**
  * A plugin that embeds into the HTTP pipeline and extends Ktor functionality.
```

Comment:
It's not entirely a plugin. It is a factory (or builder) of a plugin.
Once you call `install(factory)` (`install(builder)`) in creates a plugin instance which actually represents a plugin.

I'm not against removing the `builder` word but please let's think about making the documentation correct.
cc: @andreyaksenov

Reply:
I haven't fixed API docs yet, sorry